### PR TITLE
[WIP] Super nodes

### PR DESCRIPTION
**Pull request by vict0rsch**
* [x] verify importance of tag 0 nodes in prediction
* [x] remove all tag `0` nodes
* [x] one super node per graph
* [x] one supernode per atom type
* [x] one supernode per atom type with min distance
* [x] positional embeddings

### CODE REVIEW

File: get_data_sample.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        # super node position for a batch is the mean of its aggregates
+        sn_pos = [b.pos[sub_nodes[i]].mean(0) for i in range(batch_size)]
+        # target relaxed position is the mean of the super node's aggregates
+        # (per batch)
+        sn_pos_relaxed = [
+            b.pos_relaxed[sub_nodes[i]].mean(0) for i in range(batch_size)
+        ]
+        # the force applied on the super node is the mean of the force applied
+        # to its aggregates (per batch)
+        sn_force = [b.force[sub_nodes[i]].mean(0) for i in range(batch_size)]
```
Comment (AlexDuvalinho) [inline get_data_sample.py:129]: Same, useless for tag0 nodes.

File: get_data_sample.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        )
+        data.natoms = data.ptr[1:] - data.ptr[:-1]
+
+        # number of aggregated nodes into the super node, per batch
+        data.sn_nodes_aggregates = tensor([len(s) for s in sub_nodes], device=device)
+        # super node position for a batch is the mean of its aggregates
+        sn_pos = [b.pos[sub_nodes[i]].mean(0) for i in range(batch_size)]
+        # target relaxed position is the mean of the super node's aggregates
+        # (per batch)
+        sn_pos_relaxed = [
```
[the remark this answers is not in the document]
Comment (vict0rsch) [reply get_data_sample.py:124]: indeed ; very small overhead though

File: get_data_sample.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        # the force applied on the super node is the mean of the force applied
+        # to its aggregates (per batch)
+        sn_force = [b.force[sub_nodes[i]].mean(0) for i in range(batch_size)]
+
+        # per-atom tensors
+
+        # SNs are last in their batch
+        data.atomic_numbers = cat(
+            [
+                cat([b.atomic_numbers[non_sub_nodes[i]], tensor([-1], device=device)])
```
Comment (vict0rsch) [inline get_data_sample.py:136]: @AlexDuvalinho I put `-1` as super nodes' atomic number but it should probably be something else. Especially for embedding lookups. What's the max?

Comment (AlexDuvalinho) [reply get_data_sample.py:136]: 83 is the max in the dataset. So you can safely put 84 if desired. Most models have embedding lookup table of size 85 so it shall be fine.

**Pull request merged**